tool/src/org/antlr/v4/codegen/model/SerializedJavaATN.java

```diff
@@ -0,0 +1,37 @@
+package org.antlr.v4.codegen.model;
+
+import org.antlr.v4.codegen.OutputModelFactory;
+import org.antlr.v4.codegen.Target;
+import org.antlr.v4.runtime.atn.ATN;
+import org.antlr.v4.runtime.atn.ATNSerializer;
+import org.antlr.v4.runtime.misc.IntegerList;
+
+/** A serialized ATN for the java target, which requires we use strings and 16-bit unicode values */
+public class SerializedJavaATN extends SerializedATN {
+	private final String[] serializedAsString;
+	private final String[][] segments;
+
+	public SerializedJavaATN(OutputModelFactory factory, ATN atn) {
```

Comment:
I suggest using universal class for SerializedATN. Is it possible to use `Object[]` array both for `String[]` and for `Integer[]`?

If not it probably makes sense to rename the classes to `SerializedATNByInt` and `SerializedATNByString`.

Reply:
yeah, I wasn't sure about this one but given that it's a very different encoding it's probably ok to have separate objects in the model because that leads to a separate output template.